llvm/lib/Transforms/InstCombine/InstCombineAtomicRMW.cpp

```diff
@@ -68,6 +68,12 @@ bool isSaturating(AtomicRMWInst& RMWI) {
     case AtomicRMWInst::FMin:
       // minnum(x, -inf) -> +inf
       return CF->isNegative() && CF->isInfinity();
+    case AtomicRMWInst::FMaximum:
+      // maximum(x, +inf) -> +inf
+      return !CF->isNegative() && CF->isInfinity();
+    case AtomicRMWInst::FMinimum:
+      // minimum(x, -inf) -> +inf
+      return CF->isNegative() && CF->isInfinity();
```

Comment:
I don't think these are correct? Inf saturates maxnum, NaN saturates maximum.

Reply:
Thanks. I've removed these, as @arsenm says below they should be done in a later patch.